### PR TITLE
Configure basic networking based on OVF environment.

### PR DESCRIPTION
**Pull request by robinelfrink**
* Adapt OVF XML spec from official 1.1.8 release.
* Rename disk image and OVA file.
* Add address and gateway properties to the OVF spec.
* Configure basic networking based on OVF environment.

### CODE REVIEW

File: roles/vmware-ova/ovfconf/files/ovfconf [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+ADDRESS=`/usr/bin/vmtoolsd --cmd "info-get guestinfo.ovfenv" | /bin/grep 'key="address"' | /bin/sed 's/.*value="\([^"]*\)".*/\1/'`
+GATEWAY=`/usr/bin/vmtoolsd --cmd "info-get guestinfo.ovfenv" | /bin/grep 'key="gateway"' | /bin/sed 's/.*value="\([^"]*\)".*/\1/'`
+
+if [ ! -z "${ADDRESS}" ]; then
+    /bin/sed -i "s~address dhcp~address ${ADDRESS}~" /opt/vyatta/etc/config/config.boot
+fi
+if [ ! -z "${GATEWAY}" ]; then
+    cat <<EOF >>/opt/vyatta/etc/config/config.boot
```
Comment (dmbaturin) [inline roles/vmware-ova/ovfconf/files/ovfconf]: We should not be using this approach. When it was used, it was for lack of better tools for the job. The new vyos.configtree library is already used in some migration scripts, you can find examples in https://github.com/vyos/vyos-1x/blob/current/src/migration-scripts/system/7-to-8 and https://github.com/vyos/vyos-1x/blob/current/src/migration-scripts/config-management/0-to-1

File: roles/vmware-ova/ovfconf/files/ovfconf [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+ADDRESS=`/usr/bin/vmtoolsd --cmd "info-get guestinfo.ovfenv" | /bin/grep 'key="address"' | /bin/sed 's/.*value="\([^"]*\)".*/\1/'`
```
Comment (dmbaturin) [inline roles/vmware-ova/ovfconf/files/ovfconf]: This is a very fragile way to parse XML. We should be using a language for which actual XML parsers are available, in our case, Python.

File: roles/vmware-ova/ovfconf/files/ovfconf.service [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+Description=Configure initial address based on OVF environment
+After=vmtoolsd.service syslog.target
+Wants=network-pre.target
+Before=network-pre.target vyatta-router.service
+ConditionVirtualization=vmware
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+ExecStart=/usr/local/sbin/ovfconf
```
Comment (dmbaturin) [inline roles/vmware-ova/ovfconf/files/ovfconf.service]: The script should not be outside of the image.
I've made a prototype of a config modification script: https://github.com/vyos/vyos-1x/blob/current/src/utils/initial-setup
Common functionality can be moved to a module, and then we can produce two scripts, one for OVF first boot, and another for interactive installation.

**Review by dmbaturin — CHANGES_REQUESTED** [review]

File: roles/vmware-ova/ovfconf/files/ovfconf.service (hunk 3 above)
Comment (robinelfrink) [reply roles/vmware-ova/ovfconf/files/ovfconf.service]: This would require changes to the `vyos-1.x`-repository, right?

**Pull request closed without merging**